ui/src/utils/PromptApi.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO: merge with UtteranceApiService.js (in Ashley's PR)
+
+class PromptApi {
+    constructor() {}
+
+    getNewPrompt() {
+        return fetch('/prompt')
+            .then((resp) => resp.json())
+            .then((prompt) => {
+                if (Object.keys(prompt).length === 0) {
+                    return {
+                        type: 'EMPTY',
+                        body: '',
+                    };
+                } else {
+                    return prompt;
+                }
+            });
+    }
+
+    resetAllPromptsUnread() {
+        return fetch('/prompt/all', {method: 'POST'}).then((resp) =>
+            resp.body()
+        );
+    }
+}
+
+const promptApi = new PromptApi();
+
+export default promptApi;
```

Comment:
What functionality do these last two lines add?

Reply:
Looks like we don't export the class definition, instead create a singleton instance `promptApi` and export that instead. Is that the approach here?

Reply:
yes; this just ensures that only one instance of the promptApi exists across the app. right now this just avoids you having to write `const propmtApi = new PromptApiService()` in each file you use it (which is why I did that), but it's more commonly used if the class needed to hold one state across the app

Reply:
nvm, moved to functions as per winnie's suggestion